typegraph/core/src/conversion/runtimes.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::errors::Result;
+use crate::global_store::Store;
+use crate::runtimes::{Materializer as RawMaterializer, MaterializerData, Runtime};
+use crate::{typegraph::TypegraphContext, wit::runtimes::Effect as WitEffect};
+use common::typegraph::{Effect, EffectType, Materializer, TGRuntime};
+use indexmap::IndexMap;
+use serde_json::json;
+
+fn effect(typ: EffectType, idempotent: bool) -> Effect {
+    Effect {
+        effect: Some(typ),
+        idempotent,
+    }
+}
+
+fn convert_effect(eff: WitEffect) -> Effect {
+    match eff {
+        WitEffect::None => effect(EffectType::None, true),
+        WitEffect::Create(idemp) => effect(EffectType::Create, idemp),
+        WitEffect::Update(idemp) => effect(EffectType::Update, idemp),
+        WitEffect::Delete(idemp) => effect(EffectType::Delete, idemp),
+        WitEffect::Upsert(idemp) => effect(EffectType::Upsert, idemp),
+    }
+}
+
+pub fn convert_materializer(
+    c: &mut TypegraphContext,
+    s: &Store,
+    mat: &RawMaterializer,
+) -> Result<Materializer> {
+    match &mat.data {
```

Comment:
I wonder the [enum_dispatch](https://github.com/metatypedev/metatype/blob/main/meta-cli/src/cli/mod.rs#L82) pattern may not be easier than a single convert function